src/EditorFeatures/Core/Implementation/Classification/SemanticClassificationTaggerProvider.cs

```diff
@@ -56,13 +56,33 @@ internal partial class SemanticClassificationTaggerProvider : AsynchronousTagger
             _typeMap = typeMap;
         }
 
-        protected override ITaggerEventSource CreateEventSource(ITextView textViewOpt, ITextBuffer subjectBuffer)
+        protected override ITaggerEventSource CreateEventSource(ITextView textView, ITextBuffer subjectBuffer)
         {
+            this.AssertIsForeground();
+            const TaggerDelay Delay = TaggerDelay.Short;
+
             // Note: we don't listen for OnTextChanged.  Text changes to this this buffer will get
             // reported by OnSemanticChanged.
             return TaggerEventSources.Compose(
-                TaggerEventSources.OnSemanticChanged(subjectBuffer, TaggerDelay.Short, _semanticChangeNotificationService),
-                TaggerEventSources.OnDocumentActiveContextChanged(subjectBuffer, TaggerDelay.Short));
+                TaggerEventSources.OnViewSpanChanged(textView, Delay),
+                TaggerEventSources.OnSemanticChanged(subjectBuffer, Delay, _semanticChangeNotificationService),
+                TaggerEventSources.OnDocumentActiveContextChanged(subjectBuffer, Delay));
+        }
+
+        protected override IEnumerable<SnapshotSpan> GetSpansToTag(ITextView textView, ITextBuffer subjectBuffer)
+        {
+            this.AssertIsForeground();
```

Comment:
@CyrusNajmabadi I agree that we don't need real semantic classification in the scroll map -- since it groups lines it's unlikely you'd even see a difference if you tried. What prevents this from running inside of scroll map though?